addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -311,38 +299,39 @@ function factory(dependencies) {
                 // knowledge of the channel
                 return;
             }
-            if (channel.channel_type === 'channel') {
-                // disabled on `channel` channels for performance reasons
-                return;
-            }
             const lastMessage = this.env.models['mail.message'].insert({id: last_message_id});
-            const updateData = {
-                partnerSeenInfos: [['insert',
-                    {
-                        id,
-                        lastFetchedMessage: [['link', lastMessage]],
-                        lastSeenMessage: [['link', lastMessage]],
-                        partner: [['insert', {id: partner_id}]],
-                    }],
-                ],
-                messageSeenIndicators: [['insert',
-                    {
-                        id: this.env.models['mail.message_seen_indicator'].computeId(last_message_id, channel.id),
-                        message: [['link', lastMessage]],
-                    }
-                ]],
-            };
+            // restrict computation of seen indicator for "non-channel" channels
+            // for performance reasons
+            const shouldComputeSeenIndicators = channel.channel_type !== 'channel';
+            const updateData = {};
+            if (shouldComputeSeenIndicators) {
+                this.env.models['mail.thread_partner_seen_info'].insert({
+                    channelId: channel.id,
+                    lastSeenMessage: [['link', lastMessage]],
+                    partnerId: partner_id,
+                });
+                Object.assign(updateData, {
+                    messageSeenIndicators: [['insert',
+                        {
+                            id: this.env.models['mail.message_seen_indicator'].computeId(last_message_id, channel.id),
```

Comment:
Maybe but I think it's kind of out of scope of this PR as it did not intend to change indicators (this line is in diff but no because of a change of messageSeenIndicators). We should create a technical task to change these I think.

Reply:
Ok to create a new task, I don't want to block this one further.

Reply:
https://www.odoo.com/web#id=2335647&action=333&active_id=1750&model=project.task&view_type=form&cids=1&menu_id=4720